### DISCUSSION TITLE
Handling FlatList re-rendering when using Mobx to store nested data objects

### DISCUSSION THREAD

**Discussion by pratikdhody** [mobxjs/mobx#3626] [Q&A] [answered]
Hi,

I have a flatlist showing items and the data for this is stored in a mobx store.
I've provided the functionality for users to be able to filter via searching with text on a few of the fields.
The search functionality works but I'm running into issues where the Flatlist is not re-rendering.
Console logging the store shows correct number of items based on the search filter.

I've tried to .slice() the store but this hasn't worked. I've tried to js() the store as the data prop is ReadOnly but that hasn't worked either. I've also added <Observer> component to the renderItem and I've also tried store.nestedstore.length on extraData and that hasn't worked either. I think the reason for this is because the store itself has nested arrays etc.

Does anyone else have any other ideas on how to re-render flatlists when using Mobx store and where the store itself is being mutated?

**Comment by mweststrate** [comment, marked as answer]
The above doesn't contain nearly enough information to enable us to provide an answer. There is no code, "Flatlist" is a meaningless term to us, there is nothing runnable insight. Please file an issue instead and follow the template if something isn't working as expected.

**Reply by pratikdhody** [reply]
Okie dokie. https://github.com/mobxjs/mobx/issues/3627 has been raised
